engine/access/rest/rest_api.go

```diff
@@ -0,0 +1,94 @@
+package rest
+
+import (
+	"errors"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/engine/access/rest/generated"
+	"github.com/onflow/flow-go/engine/access/rpc/backend"
+	"github.com/onflow/flow-go/model/encoding"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+)
+
+// RestAPIHandler provides the implementation of each of the REST API
+type RestAPIHandler struct {
+	backend *backend.Backend
+	logger  zerolog.Logger
+	encoder encoding.Encoder
+}
+
+func NewRestAPIHandler(backend *backend.Backend, logger zerolog.Logger) *RestAPIHandler {
+	return &RestAPIHandler{
+		backend: backend,
+		logger:  logger,
+		encoder: encoding.DefaultEncoder, //use the default JSON encoder
+	}
+}
+
+func (restAPI *RestAPIHandler) BlocksIdGet(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+	vars := mux.Vars(r)
+	idParam, ok := vars["id"]
+	if !ok {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	// gorilla mux retains opening and ending square brackets for ids
+	idParam = strings.TrimSuffix(idParam, "]")
+	idParam = strings.TrimPrefix(idParam, "[")
+
+	ids := strings.Split(idParam, ",")
+
+	blocks := make([]*generated.Block, len(ids))
```

Comment:
do we have a max number of IDs? we should enforce some sensible default max.

Reply:
Yeah, I second this. It would also be a good idea to add this to the spec as well.

Reply:
1. Added to the spec - [PR](https://github.com/onflow/flow/pull/686)
2. Added a validation here in the server as well. Unfortunately, the generated server code (unlike the generated client code) does not add the validation automatically.